kubernetes/policies/general/deprecated_api_version.rego

```diff
@@ -0,0 +1,57 @@
+package appshield.kubernetes.KSV101
+
+import data.lib.kubernetes
+import data.lib.utils
+
+__rego_metadata__ := {
+	"id": "KSV101",
+	"title": "Using A Deprecated API Version",
+	"version": "v1.0.0",
+	"severity": "Critical",
+	"type": "Kubernetes Security Check",
+	"description": "Check if any objects are using a deprecated version of API.",
+	"recommended_actions": "Don't use deprecated API versions",
+}
+
+recommendedVersions := {
+	"extensions/v1beta1": {
+		"Deployment": "apps/v1",
+		"DaemonSet": "apps/v1",
+		"Ingress": "apps/v1",
+	},
+	"apps/v1beta1": {
+		"Deployment": "apps/v1",
+		"StatefulSet": "apps/v1",
+	},
+	"apps/v1beta2": {
+		"Deployment": "apps/v1",
+		"DaemonSet": "apps/v1",
+		"StatefulSet": "apps/v1",
+	},
+}
+
+# Get all containers which use deprecated api versions
+getDeprecatedApi[name] {
+	allContainers := kubernetes.containers[_]
```

Comment:
containers is not relevant in this policy. some resources doesn't have containers